web/js/palettes/legend.js

```diff
@@ -58,6 +58,7 @@ export function palettesLegend(spec) {
 
   var renderScale = function ($legendPanel, legend, index, layerId) {
     var $container = $('<div></div>')
+      .attr('id', legend.id)
```

Comment:
@ZachTRice You might want to mix the layer-Id with the legend-Id to create a unique identifier. I think some layers use the same legend and when more than one of these layers are present in the sidebar there can be problems selecting the correct one.

Reply:
@Benjaki2 the legend.id is a unique identifier. This value is what is in the config/palettes/[layer].json id field + an index count starting from 0 + '_legend', so even layers with the same layerId will use this as a way to uniquely identify each legend. 
![screenshot_17](https://user-images.githubusercontent.com/17348884/41468586-83f9e498-7078-11e8-8561-6fa3d6627c23.png)